docs/design.md

```diff
@@ -0,0 +1,59 @@
+# OSBA Design
+
+We can roughly split OSBA's architecture into three pieces:
+
+1.  REST API frontend
+2.  Stable Storage
+3.  Asynchronous processing
```

Comment:
I think maybe call out the modules here? This design doc is probably for people that will eventually contribute modules so i think it would be good to add a section detailing what the module does vs what the front end does.

Reply:
Sounds good! I'll add a fourth item here explaining how they interact with the above 3